GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -320,6 +327,98 @@ public void run() {
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render +y face
+        centerCameraTransform.rotateByAxis(-90, 0, 1, 0);
+        centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render -y face
+        centerCameraTransform.rotateByAxis(180, 1, 0, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render +z face
+        centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+        centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render -z face
+        centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        centerCameraObject.detachCamera();
+        mainCameraRig.getOwnerObject().removeChildObject(centerCameraObject);
+    }
+
+    void returnScreenshot3DToCaller(final GVRScreenshot3DCallback callback,
+            final List<byte[]> byteArrayList, final int width, final int height) {
+        if (byteArrayList.size() != 6) {
+            throw new IllegalArgumentException("byteArrayList length is not 6.");
+        } else {
+            // run the callback function in a background thread
+            Threads.spawn(new Runnable() {
+                public void run() {
+                    List<Bitmap> bitmapList = new ArrayList<Bitmap>();
```

Comment:
That would be even more efficient, though the `List` gives you enough extra functionality that it may be worth it. Both this and my original comment are pretty small issues, though.